### PR TITLE
Redirect _l function to non-locale version

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Since locale can't be really changed in NuttX, the locale version is same as non-locale version.

- locale: Add LC_GLOBAL_LOCALE which is specified by the standard
- libc: Implement local_t related functions

## Impact
New API

### CODE REVIEW

**Comment by gustavonihei** [comment]
LGTM, but CI is failing on the **libc++** build.

**Comment by xiaoxiang781216** [comment]
> LGTM, but CI is failing on the **libc++** build.

It's due to the macro expansion don't understand C++ namespace, so I change all macros to functions.

File: libs/libc/string/lib_strcoll.c [hunk 1]
```diff
@@ -56,4 +56,9 @@ int strcoll(const char *a, const char *b)
 {
   return strcmp(a, b);
 }
+
+int strcoll_l(const char *a, const char *b, locale_t l)
```
Comment (gustavonihei) [inline libs/libc/string/lib_strcoll.c:60]:
```suggestion
int strcoll_l(FAR const char *a, FAR const char *b, locale_t l)
```
Missing `FAR` qualifier.

File: libs/libc/wchar/lib_wcsftime.c [hunk 2]
```diff
@@ -39,4 +39,10 @@ size_t wcsftime(FAR wchar_t *s, size_t maxsize, FAR const wchar_t *format,
 
   return strftime((FAR char *)s, maxsize, (FAR char *)format, tim_p);
 }
+
+size_t wcsftime_l(FAR wchar_t *s, size_t maxsize, FAR const wchar_t *format,
+                 FAR const struct tm *tim_p, locale_t l)
```
Comment (gustavonihei) [inline libs/libc/wchar/lib_wcsftime.c:44]:
```suggestion
                  FAR const struct tm *tim_p, locale_t l)
```
nit: alignment

Comment (xiaoxiang781216) [reply libs/libc/wchar/lib_wcsftime.c]: Done.

File: libs/libc/string/lib_strcoll.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libc/string/lib_strcoll.c]: Done.

File: libs/libc/string/lib_strncasecmp.c [hunk 3]
```diff
@@ -50,4 +50,9 @@ int strncasecmp(const char *cs, const char *ct, size_t nb)
 
   return result;
 }
+
+int strncasecmp_l(const char *cs, const char *ct, size_t nb, locale_t l)
```
Comment (gustavonihei) [inline libs/libc/string/lib_strncasecmp.c:54]:
```suggestion
int strncasecmp_l(FAR const char *cs, FAR const char *ct, size_t nb, locale_t l)
```
Also missing FAR qualifier

File: libs/libc/string/lib_strncasecmp.c [hunk 4]
```diff
@@ -50,4 +50,9 @@ int strncasecmp(const char *cs, const char *ct, size_t nb)
...
+{
+  return strncasecmp(cs, ct, l);
```
Comment (gustavonihei) [inline libs/libc/string/lib_strncasecmp.c:56]:
```suggestion
  return strncasecmp(cs, ct, nb);
```
Wrong parameter passed. CI build is failing due to this error.

File: libs/libc/string/lib_strncasecmp.c [hunk 5]
```diff
@@ -50,4 +50,9 @@ int strncasecmp(const char *cs, const char *ct, size_t nb)
 
   return result;
 }
+
+int strncasecmp_l(FAR const char *cs, FAR const char *ct, size_t nb, locale_t l)
```
Comment (gustavonihei) [inline libs/libc/string/lib_strncasecmp.c:54]: @xiaoxiang781216 Now we have an nxstyle error :rofl:

Comment (xiaoxiang781216) [reply libs/libc/string/lib_strncasecmp.c]: Done.

File: libs/libc/string/lib_strncasecmp.c (hunk 4 above)
Comment (xiaoxiang781216) [reply libs/libc/string/lib_strncasecmp.c]: Done.

File: libs/libc/string/lib_strncasecmp.c (hunk 3 above)
Comment (xiaoxiang781216) [reply libs/libc/string/lib_strncasecmp.c]: Done.

**Review by gustavonihei — APPROVED** [review]
LGTM

File: include/wchar.h [hunk 6]
```diff
@@ -34,6 +34,7 @@
 
 #include <stdio.h>
 #include <stddef.h>
+#include <local.h>
```
Comment (gustavonihei) [inline include/wchar.h:37]:
```suggestion
#include <locale.h>
```
```
/github/workspace/sources/nuttx/include/wchar.h:37:10: fatal error: local.h: No such file or directory
   37 | #include <local.h>
```

Comment (xiaoxiang781216) [reply include/wchar.h]: Done, thanks.

**Comment by gustavonihei** [comment]
@xiaoxiang781216 Will you submit this patch to LLVM project? If so, could you please post the link once you do it?

**Comment by xiaoxiang781216** [comment]
Yes, I will prepare the patch to llvm.

**Pull request merged**